api/processors/auxil/algorithms.py

```diff
@@ -0,0 +1,171 @@
+# DO NOT USE IN PROD, WORK IN PROGRESS
+
+from collections.abc import Iterable
+from dataclasses import dataclass
+from datetime import datetime
+
+from django.utils import timezone
+
+from api.models import AgeRange, Group, Student, Teacher
+from api.models.auxil.constants import (
+    DEFAULT_LESSON_DURATION_MIN,
+    MAX_AGE_KIDS_GROUP,
+    MAX_AGE_TEEN_GROUP,
+)
+from api.models.auxil.status_setter import StatusSetter
+from api.models.choices.communication_language_mode import CommunicationLanguageMode
+from api.models.choices.log_event_type import (
+    GroupLogEventType,
+    StudentLogEventType,
+    TeacherLogEventType,
+)
+from api.models.choices.status import (
+    GroupProjectStatus,
+    StudentSituationalStatus,
+    TeacherProjectStatus,
+    TeacherSituationalStatus,
+)
+from api.models.language_and_level import LanguageAndLevel
+from api.processors.auxil.log_event_creator import GroupLogEventCreator
+
+
+class GroupBuilderAlgorithm:
+    @dataclass
+    class GroupSizeRestriction:
```

Comment:
If that's a project convention overall, then ok. I personally believe if something is a detail of the class implementation and not intended to be used outside, it shouldn't be exposed on a module level to avoid confusion.

Reply:
Yeah, @Roxe322 ruthlessly imposed this convention on us! 😆 Let's keep the classes at module level, then. It's not a big deal, especially when we list the classes to import in `__init__.py` and its `__all__`.

Reply:
Yes, why not! Did this already